packages/core/src/typeschema/types.ts

```diff
@@ -157,12 +158,17 @@ class StructureDefinitionParser {
         this.checkFieldEnter(element, field);
 
         // Record field in schema
-        if (this.backboneContext && element.path?.startsWith(this.backboneContext.path + '.')) {
-          this.backboneContext.type.fields[elementPath(element, this.backboneContext.path)] = field;
-        } else if (this.backboneContext?.parent && element.path?.startsWith(this.backboneContext.parent.path + '.')) {
-          this.backboneContext.parent.type.fields[elementPath(element, this.backboneContext.parent.path)] = field;
-        } else {
-          const path = elementPath(element, this.resourceSchema.name);
+        let parentContext: BackboneContext | undefined = this.backboneContext;
+        while (parentContext) {
+          if (element.path?.startsWith(parentContext.path + '.')) {
+            parentContext.type.fields[elementPath(element, parentContext.path)] = field;
+            break;
+          }
+          parentContext = parentContext.parent;
+        }
+
+        if (!parentContext) {
+          const path = elementPath(element, this.root.path);
```

Comment:
The change from `elementPath(element, this.resourceSchema.name)`

to `elementPath(element, this.root.path)`

is not directly related to this PR (my bad), but still a change that we need to make in https://github.com/medplum/medplum/pull/2859

Within R4 `StructureDefinition`s, there are 2 cases where `StructureDefinition.name !== ElementDefinition.path`.  For `SimpleQuantity` and `MoneyQuantity`, the names are the names, but the root `ElementDefinition.path` is `Quantity`.  I guess they're kind of like mini profiles within the R4 spec?  Either way, we need to use `StructureDefinition.name` for the type name, and `ElementDefinition.path` for the path.

Reply:
Would you mind adding this as a comment?  The context is super helpful around these kinds of edge cases in the code